crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -37,6 +39,57 @@ impl DynamicScene {
         Self::from_world(&scene.world, type_registry)
     }
 
+    /// Create a Dynamic Scene with specific entities.
+    ///
+    /// The generic parameter is used as a `Query` filter.
+    ///
+    /// The created scene will include only the entities that match the query
+    /// filter provided. All components that impl `Reflect` will be included.
+    pub fn from_query_filter<F>(world: &mut World) -> Self
+    where
+        F: ReadOnlyWorldQuery + 'static,
+    {
+        let mut ss = SystemState::<Query<Entity, F>>::new(world);
+
+        let type_registry = world
+            .get_resource::<AppTypeRegistry>()
+            .expect("The World provided for scene generation does not contain a TypeRegistry")
+            .read();
+
+        let q = ss.get(world);
+
+        let entities = q
+            .iter()
+            .map(|entity| {
+                let get_reflect_by_id = |id| {
```

Comment:
So despite copying this function over, I actually am not 100 percent positive I know how it works haha. It seems like the function is getting two lists of components and then doing a filter_map below to get only the serializable components on each entity. 

But I am a little confused by the difference between get_reflect_by_id and components. Isn't get_reflect_by_id also a bunch of components?